hypothesis-python/src/hypothesis/internal/conjecture/providers.py

```diff
@@ -371,8 +412,10 @@ class HypothesisProvider(PrimitiveProvider):
 
     def __init__(self, conjecturedata: Optional["ConjectureData"], /):
         super().__init__(conjecturedata)
-        self.local_constants = _get_local_constants()
         self._random = None if self._cd is None else self._cd._random
+        self.local_constants = (
+            None if self._random is None else _get_local_constants(self._random)
+        )
```

Comment:
We could further reduce the best-case cost by making this a cached property; then if user code doesn't happen to draw from the provider (explicit phase, or replaying and shrinking a failing example without the explain phase) we don't bother collecting constants.

------------

Unrelated except that I just spotted it, we have some stale comments and docstrings about `generate_novel_prefix` describing the buffer etc.  Should be a quick cleanup at some point?